build.mk

```diff
@@ -7,11 +7,14 @@ rwildcard=$(foreach d,$(wildcard $1*), \
 SRCFILES=$(call rwildcard,$(SRCDIR)/,*.c *.S)
 SRCFILES=$(call rwildcard,$(SRCDIR)/,*.c *.S)
 
-TAG_ID=$(shell git -C $(SRCDIR) tag --points-at HEAD)
-REV_ID=$(shell git -C $(SRCDIR) rev-parse --short HEAD)
-
+-TAG_ID?=$(shell git -C $(SRCDIR) tag --points-at HEAD)
```

Comment:
remove first '-'

Reply:
I missing this point 'Suppress error output when git command failed to execute', but it looks doesn't work, we always got 'unknown' even through git command successful executed.

Reply:
```makefile
ifeq ($(REV_ID),)
  REV_ID=$(shell git -C $(SRCDIR) tag --points-at HEAD)
  ifeq ($(REV_ID),)
    REV_ID=$(shell git -C $(SRCDIR) rev-parse --short HEAD)
  endif
  ifeq ($(REV_ID),)
    REV_ID=unknown
  endif
endif
VERSION_CFLAGS=-DCOMMIT_ID=\"$(REV_ID)\"
```

openwrt pkg build with

```makefile
MAKE_FLAGS += REV_ID="$(PKG_SOURCE_VERSION)"
```

What do you think?

@1715173329 @ysc3839

Reply:
> I missing this point 'Suppress error output when git command failed to execute', but it looks doesn't work, we always got 'unknown' even through git command successful executed.

Yes, `-` did suppress error output, but I guess it should be only used with common shell commands.

You solution looks great, please commit it, thank you!